File: Backend/src/models/user_organisations.py
# Backend/src/models/user_organisations.py
# Definition of the UserOrganisation model, connecting users and organisations

# Author: Valentin Haas, 2024

# Library imports
from sqlalchemy import Column, Integer

# Project imports
from ._base import BaseTable


class UserOrganisationsTable(BaseTable):
    """Definition of the UserOrganisation model for the database."""

    __tablename__ = "UserOrganisations"

    User = Column(Integer, primary_key=True, foreign_key="Users.Id")
    """The unique identifier of the user."""

    Organisation = Column(Integer, primary_key=True, foreign_key="Organisations.Id")
    """The unique identifier of the organisation."""

File: Backend/src/models/userorganisations.go
// Backend/src/models/userorganisations.go
// Definition of the UserOrganisation model, connecting users and organisations

// Author: Valentin Haas, 2025
package models

import "time"

// UserOrganisation defines the relationship between users and organisations.
type UserOrganisation struct {
	User           User         // Referenced User Object
	UserID         uint         `gorm:"primaryKey"` // The user in the relationship (foreign key reference to User.ID).
	Organisation   Organisation // Referenced Organisation Object
	OrganisationID uint         `gorm:"primaryKey"` // The organisation in the relationship (foreign key reference to Organisation.ID).
	CreatedAt      time.Time    // Time the model was created. Auto Populated by Gorm.
	UpdatedAt      time.Time    // Time the model was updated. Auto Populated by Gorm.
	DeletedAt      time.Time    // Time the model was deleted. Auto Populated by Gorm.
}
